### PR TITLE
RC-350 Unify ruling expectation file locations.

### PR DESCRIPTION
**Pull request by romainbrenguier**
## Summary
- Move ruling expectation files from `its/ruling/src/test/expected/<project>/<language>-<ruleId>.json` to `its/ruling/src/test/resources/expected/<project>/<ruleId>.json`
- Update Java ruling test (LITS-based) to merge per-language expectation dirs into a flat directory at build time
- Update JS/TS ruling pipeline (`lits.ts`, `testProject.ts`) to write and compare results using the new layout
- Update CI scripts (`sync-results.mjs`, `generate-report.mjs`, `build.yml`) to use the new paths

## Motivation
Normalize ruling expectation file locations so that it is consistent across all analyzers.

## Test plan
- [x] JS/TS ruling tests pass (`npm run ruling`)
- [ ] Java ruling tests pass (`mvn test -Dtest=RulingTest`)
- [ ] `lits.test.ts` unit tests pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Agent workflow

Tool link: Tool link: https://github.com/SonarSource/languages-experimental-tooling/tree/romain/my-tickets/personal/romain-brenguier

Iterated on the PR with `uv run ci_loop.py` for 3 iterations. 
:heavy_check_mark: The PR is now ready for review.

### CODE REVIEW

File: .claude/skills/ruling/SKILL.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ sh tools/ruling-debug-script.sh
 Results:
```
Comment (guillemsarda) [inline .claude/skills/ruling/SKILL.md:32]: For file `packages/ruling/lits.ts`:

Could we keep creating `actualPath` unconditionally in `writeResults()`? With mkdir inside `writeIssues()`, a project that produces zero issues (including parsing errors) leaves no actual-results directory, so dir-compare throws ENOENT instead of comparing the results. This doesn't affect the current projects, but could affect a future zero-issue project even if its expected directory exists. Keeping the directory creation here and adding an empty-results test would cover that edge case.

File: .claude/skills/ruling/SKILL.md [hunk 2]
```diff
@@ -44,13 +44,13 @@ mvn verify -Dtest=RulingTest -Dmaven.test.redirectTestOutputToFile=false
 Copy actual to expected:
```
Comment (guillemsarda) [inline .claude/skills/ruling/SKILL.md:44]: Could we keep `target/actual/` as the source in the copy command and as the comparison directory in the `diff` command? `RulingTest.java` (lines 280-282) still sets `sonar.lits.dump.new` to `target/actual/<project>`, so changing these instructions to `target/expected` makes them point to the wrong output directory. Only the baseline path needs to change to `src/test/resources/expected/`.

**Review by guillemsarda — APPROVED** [review]
Approving with 2 minor concerns.

File: .claude/skills/ruling/SKILL.md (hunk 2 above)
Comment (romainbrenguier) [reply .claude/skills/ruling/SKILL.md:44]: That's not changed by this PR, it was already packages/ruling/actual before, and it is still what the github action use to update the ruling expectations.
